crates/wit-parser/src/ast/resolve.rs

```diff
@@ -422,69 +505,69 @@ impl<'a> Resolver<'a> {
         for item in world.items.iter() {
-            let (docs, name, kind, desc, spans, interfaces) = match item {
+            let (docs, kind, desc, spans, interfaces) = match item {
                 // handled in `resolve_types`
                 ast::WorldItem::Use(_) | ast::WorldItem::Type(_) => continue,
 
                 ast::WorldItem::Import(import) => (
                     &import.docs,
-                    &import.name,
                     &import.kind,
                     "import",
                     &mut import_spans,
                     &mut imported_interfaces,
                 ),
                 ast::WorldItem::Export(export) => (
                     &export.docs,
-                    &export.name,
                     &export.kind,
                     "export",
                     &mut export_spans,
                     &mut exported_interfaces,
                 ),
             };
-            let prev = used_names.insert(name.name, desc);
-            if let Some(prev) = prev {
-                return Err(Error {
-                    span: name.span,
-                    msg: format!(
-                        "{desc} `{name}` conflicts with prior {prev} of same name",
-                        name = name.name
-                    ),
+            let key = match kind {
+                ast::ExternKind::Interface(name, _) | ast::ExternKind::Func(name, _) => {
+                    let prev = used_names.insert(name.name, desc);
+                    if let Some(prev) = prev {
+                        return Err(Error {
+                            span: kind.span(),
+                            msg: format!(
+                                "{desc} `{name}` conflicts with prior {prev} of same name",
+                                name = name.name
+                            ),
+                        }
+                        .into());
+                    }
+                    WorldKey::Name(name.name.to_string())
                 }
-                .into());
-            }
-            let world_item = self.resolve_world_item(docs, name.name, document, kind)?;
+                ast::ExternKind::Path(path) => WorldKey::Interface(self.resolve_path(path)?),
+            };
+            let world_item = self.resolve_world_item(docs, kind)?;
             if let WorldItem::Interface(id) = world_item {
-                if interfaces.insert(id, name.name).is_some() {
-                    return Err(Error {
-                        span: name.span,
-                        msg: format!(
-                            "interface `{name}` cannot be {desc}ed more than once",
-                            name = name.name
-                        ),
-                    }
-                    .into());
+                if !interfaces.insert(id) {
+                    bail!(Error {
+                        span: kind.span(),
+                        msg: format!("interface cannot be {desc}ed more than once",),
```

Comment:
Ah this one's a bit tricky since there's no name itself but instead we've got a path which is a bit buried, so I was hopeful the span of the error would be enough for diagnoising.

Reply:
Span is fine if there's no better way to do it; just a believer in the message itself conveying as much relevant information as possible, with the span adding to it.